substratevm/src/com.oracle.svm.core.graal/src/com/oracle/svm/core/graal/snippets/CEntryPointSnippets.java

```diff
@@ -351,10 +351,10 @@ private static int enterIsolateMT(Isolate isolate) {
     public static int enterSnippet(IsolateThread thread) {
         Isolate isolate;
         if (MultiThreaded.getValue()) {
+            writeCurrentVMThread(thread);
```

Comment:
If someone moves this around they will break the LLVM stuff without knowing why and waste a ton of time.

The problem here is that `r15` reads in the function are floating so they get hoisted to the point before the thread is written. 

I see the following solutions: 
 1) Verify that there are no reads scheduled before non-null writes and reads after a null write in every function.
 2) In functions that write to `r15` all the reads simply become fixed:
       a) By annotating the functions that write to `r15` and simply make all register reads fixed.
       b) By making a phase that handles that.
       c) By making fixed versions of functions that access `r15` and only call them in the code. 

Given how rare is this I would just go with 1) and put a comment in the code. 

@peter-hofer WDYT?

Reply:
@loicottet could you send the graph for this problem. /cc @christianwimmer

Reply:
`r15` is only ever written in these snippets, so I wouldn't put too much effort into a sophisticated solution. Are these reads hoisted in Graal or in LLVM? Could there be some sort of barrier (using the term loosely) that we can insert here or during lowering that prevents reording the reads before here?

Reply:
> @loicottet could you send the graph for this problem. /cc @christianwimmer

![2019-06-24](https://user-images.githubusercontent.com/3634331/60190694-619ddb00-9833-11e9-8c35-bdfd35417b06.png)
The problem is node 28 gets emitted before node 27.

Reply:
I have trouble determining when exactly I should emit a FixedNode instead of a FloatingNode. The node accessing the thread pointer is created when generating `VMThreads$StatusSupport.compareAndSetNativeToJava()`, so the most I can do is limit the change to uninterruptible methods. Limiting it to this exact function feels like a gross hack, is there another way I can find out whether a node is being emitted in an entry point?

Reply:
This is fixed, right?